packages/context/src/Callback/Callback.container.js

```diff
@@ -1,63 +1,41 @@
-import { withRouter } from "react-router-dom";
-import { compose, withProps, lifecycle, withHandlers, pure } from "recompose";
+import React, { useEffect, useCallback } from 'react';
+import { withRouter } from 'react-router-dom';
 
-import { getUserManager, oidcLog } from "../Services";
-import CallbackComponent from "./Callback.component";
+import { getUserManager, oidcLog, withServices } from '../Services';
+import CallbackComponent from './Callback.component';
 
-let _isCalled = false;
-const _setCalling = val => {
-  _isCalled = val;
-};
-
-export const onRedirectSuccess = ({ history }) => user => {
-  oidcLog.info("Successfull Callback");
+export const onRedirectSuccess = (history, oidcLogInternal) => user => {
+  oidcLogInternal.info('Successfull login Callback');
   if (user.state.url) {
     history.push(user.state.url);
   } else {
-    oidcLog.warn("no location in state");
+    oidcLogInternal.warn('no location in state');
   }
 };
 
-export const onRedirectError = ({ history }) => error => {
-  const { message } = error;
-  oidcLog.error(
-    `There was an error handling the token callback: ${error.message}`
-  );
-  history.push(`/authentication/not-authenticated?message=${message}`);
+export const onRedirectError = (history, oidcLogInternal) => ({ message }) => {
+  oidcLogInternal.error(`There was an error handling the token callback: ${message}`);
+  history.push(`/authentication/not-authenticated?message=${encodeURIComponent(message)}`);
 };
 
-export const componentDidMountFunction = async (props, isCalled, setCall) => {
-  try {
-    oidcLog.info("Callback Container Mount");
-    let user = await props.userManager.getUser();
-    if (!isCalled && (!user || user.expired)) {
-      setCall(true);
-      user = await props.userManager.signinRedirectCallback();
-      props.onRedirectSuccess(user);
-    }
-  } catch (error) {
-    props.onRedirectError(error);
-  }
+export const CallbackContainerCore = ({
+  history,
+  getUserManager: getUserManagerInternal,
+  oidcLog: oidcLogInternal,
+}) => {
+  const onSuccess = useCallback(onRedirectSuccess(history, oidcLogInternal), [history]);
```

Comment:
you should not add the dependencies "oidcLogInternal" and "onRedirectSuccess" ?

Reply:
I don't want to inject internal deps. Because i want to test the global feature. I prefer to keep internal call, but I eject in the same module the function to test the function themselves.